syntax = "proto3";

package items;

import "google/protobuf/empty.proto";

// Mensagem para representar um item
message ItemResponse {
  string id = 1;           // ID do item
  string name = 2;         // Nome do item
  string description = 3;  // Descrição do item
}

// Mensagem para uma lista de itens
message ItemList {
  repeated ItemResponse items = 1; // Lista de itens
}

// Requisição para criar um novo item
message NewItemRequest {
  string name = 1;         // Nome do novo item
  string description = 2;  // Descrição do novo item
}

// Requisição para deletar um item
message ItemRequest {
  string id = 1;           // ID do item
}

// Serviço para as operações gRPC
service ItemService {
  rpc GetItems(google.protobuf.Empty) returns (ItemList); // Retorna todos os itens
  rpc CreateItem(NewItemRequest) returns (ItemResponse);  // Cria um novo item
  rpc DeleteItem(ItemRequest) returns (ItemResponse);     // Deleta um item pelo ID
}

